File: client/types/baseline.ts
export interface BaselineItem {
  item_id: string;
  task_name: string | null;
  instruction: string | null;
  display_content: string | null;
  expected_output: string | null;
  response_type: string | null;
  target_phoneme: string | null;
  formula_weights: Record<string, number> | null;
  fusion_weights: Record<string, number> | null;
  wpm_range: Record<string, number> | null;
}

export interface BaselineSection {
  section_id: string;
  section_name: string;
  instructions: string | null;
  order_index: number;
  items: BaselineItem[];
}

export interface BaselineAssessment {
  baseline_id: string;
  name: string;
  domain: string;
  sections: BaselineSection[];
}

export interface AttemptResult {
  attempt_id: string;
  result: string;
  computed_score: number | null;
  word_accuracy: number | null;
  phoneme_accuracy: number | null;
  fluency_score: number | null;
  speech_rate_wpm: number | null;
  speech_rate_score: number | null;
  confidence_score: number | null;
  engagement_score: number | null;
  dominant_emotion: string | null;
  pass_fail: string | null;
  asr_transcript: string | null;
}

export interface BaselineResult {
  result_id: string;
  baseline_name: string;
  raw_score: number;
  level: string;
  assessed_on: string;
}

export interface BaselineItemDetail {
  item_id: string;
  prompt_text: string | null;
  transcript: string | null;
  word_accuracy: number | null;
  phoneme_accuracy: number | null;
  fluency_score: number | null;
  speech_rate_wpm: number | null;
  speech_rate_score: number | null;
  confidence_score: number | null;
  engagement_score: number | null;
  dominant_emotion: string | null;
  final_score: number;
  pass_fail: boolean;
  created_at: string;
}

File: server/app/schemas/baseline.py
from datetime import date
from pydantic import BaseModel
from typing import Optional


class BaselineItemOut(BaseModel):
    item_id: str
    task_name: Optional[str]
    instruction: Optional[str]
    display_content: Optional[str]
    expected_output: Optional[str]
    response_type: Optional[str]
    target_phoneme: Optional[str]
    formula_weights: Optional[dict]
    fusion_weights: Optional[dict]
    wpm_range: Optional[dict]


class BaselineSectionOut(BaseModel):
    section_id: str
    section_name: str
    instructions: Optional[str]
    order_index: int
    items: list[BaselineItemOut]


class BaselineAssessmentOut(BaseModel):
    baseline_id: str
    name: str
    domain: str
    sections: list[BaselineSectionOut]


class ItemScoreSubmit(BaseModel):
    item_id: str
    score: float


class BaselineSubmitRequest(BaseModel):
    baseline_id: str
    item_scores: list[ItemScoreSubmit]


class BaselineResultOut(BaseModel):
    result_id: str
    baseline_name: str
    raw_score: int
    level: str
    assessed_on: date


class BaselineItemDetailOut(BaseModel):
    item_id: str
    prompt_text: Optional[str]
    transcript: Optional[str]
    word_accuracy: Optional[float] = None
    phoneme_accuracy: Optional[float]
    fluency_score: Optional[float]
    speech_rate_wpm: Optional[float] = None
    speech_rate_score: Optional[float] = None
    confidence_score: Optional[float] = None
    engagement_score: Optional[float] = None
    dominant_emotion: Optional[str] = None
    final_score: float
    pass_fail: bool
    created_at: str
